### PR TITLE
feat: extend links feature for custom workflow views

### PR DESCRIPTION
**Pull request by jiachengxu**
This PR extends the current links feature(https://argoproj.github.io/argo-workflows/links/), with new scope `workflow-list`, which will be used as buttons on the top of the workflow view, and users can use them to navigate to filtered/curated view of workflows. 
This PR only contains the necessary on the backend and @rbreeze will help with the UI support.
Related #10447

### CODE REVIEW

File: pkg/apis/workflow/v1alpha1/info.go [hunk 1]
```diff
@@ -6,7 +6,7 @@ package v1alpha1
 type Link struct {
 	// The name of the link, E.g. "Workflow Logs" or "Pod Logs"
 	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
-	// "workflow", "pod", "pod-logs", "event-source-logs", "sensor-logs" or "chat"
+	// "workflow", "pod", "pod-logs", "event-source-logs", "sensor-logs", "custom-buttons" or "chat"
```
Comment (terrytangyuan) [inline pkg/apis/workflow/v1alpha1/info.go:9]: Is this used anywhere?

Comment (jiachengxu) [reply pkg/apis/workflow/v1alpha1/info.go:9]: Thank you for the review @terrytangyuan!
This `scope`  will only be used by the UI code to create buttons on the workflow view, and it is not used by anywhere for now.
@rbreeze will create UI based on the scope `custom-buttons` in the next UI PRs, and this PR only make the necessary backend change

**Comment by alexec** [comment]
”custom buttons” is rather generic. What about “workflow list”?

**Comment by jiachengxu** [comment]
> ”custom buttons” is rather generic. What about “workflow list”?

I think that makes more sense, and I just updated the PR, thanks!

File: pkg/apis/workflow/v1alpha1/info.go (hunk 1 above)
Comment (terrytangyuan) [reply pkg/apis/workflow/v1alpha1/info.go]: But the docs and manifests are also updated, which might mislead users. Does it make more sense to work on a single PR? @alexec Any thoughts?

Comment (jiachengxu) [reply pkg/apis/workflow/v1alpha1/info.go]: Personally, I feel like the `main` branch can always contain un-released or incompleted features, and most users use releases and this shouldn't affect them much. But if you think that is an issue, I can also remove the user-facing change(e.g., docs, manifests) in this PR and add them back once the feature is completed.

Comment (rbreeze) [reply pkg/apis/workflow/v1alpha1/info.go]: I agree with @jiachengxu - I think it's okay to get this merged first given that it's such a small change to the docs

**Review by rbreeze — APPROVED** [review]
LGTM!

**Review by terrytangyuan — APPROVED** [review]

File: pkg/apis/workflow/v1alpha1/info.go (hunk 1 above)
Comment (terrytangyuan) [reply pkg/apis/workflow/v1alpha1/info.go]: SGTM

**Pull request merged**